pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -0,0 +1,343 @@
+// based on the state read and what is in the ApplicationLayer.Spec.
+func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ApplicationLayer")
+
+	applicationLayer, err := getApplicationLayer(ctx, r.client)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			reqLogger.Info("ApplicationLayer object not found")
+			r.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "Error querying for Application Layer")
+		r.status.SetDegraded("Error querying for Application Layer", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	preDefaultPatchFrom := client.MergeFrom(applicationLayer.DeepCopy())
+
+	updateApplicationLayerWithDefaults(applicationLayer)
+
+	err = validateApplicationLayer(applicationLayer)
+
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		r.status.SetDegraded(err.Error(), "")
+		return reconcile.Result{}, nil
+	}
+
+	// Write the application layer back to the datastore, so the controllers depending on this can reconcile.
+	if err = r.client.Patch(ctx, applicationLayer, preDefaultPatchFrom); err != nil {
+		reqLogger.Error(err, "Failed to write defaults to applicationLayer")
+		r.status.SetDegraded("Failed to write defaults to applicationLayer", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	variant, installation, err := utils.GetInstallation(ctx, r.client)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Error(err, "Installation not found")
+			r.status.SetDegraded("Installation not found", err.Error())
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "Error querying installation")
+		r.status.SetDegraded("Error querying installation", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	managementCluster, err := utils.GetManagementCluster(ctx, r.client)
+	if err != nil {
+		reqLogger.Error(err, "Error reading ManagementCluster")
+		r.status.SetDegraded("Error reading ManagementCluster", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.client)
+	if err != nil {
+		reqLogger.Error(err, "Error reading ManagementClusterConnection")
+		r.status.SetDegraded("Error reading ManagementClusterConnection", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	if managementClusterConnection != nil && managementCluster != nil {
+		err = fmt.Errorf("having both a ManagementCluster and a ManagementClusterConnection is not supported")
+		reqLogger.Error(err, "")
+		r.status.SetDegraded(err.Error(), "")
+		return reconcile.Result{}, err
+	}
```

Comment:
I don't see these being used at all. Why do we need to get these?

Reply:
fixed in 6c8d0e4aaa28c9e011f5289a218584a313676c1c